### PR TITLE
New Extension: Recurly

### PR DESCRIPTION
**Pull request by kudrykv**
## Description

Quickly search and inspect users and users' subscriptions.

## Screencast

CleanShot 2022-07-13 at 21 33 08
CleanShot 2022-07-13 at 21 34 46
CleanShot 2022-07-13 at 21 35 27
CleanShot 2022-07-13 at 21 36 32
CleanShot 2022-07-13 at 21 36 39

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/recurly/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Recurly Changelog
+
+## [Initial Version] - 2022-05-05
+
+- `Tenants configuration` command;
+- `Accounts Search By Code` command;
+- `Accounts Search by Email` command.
```
Comment (danielsequeira) [inline extensions/recurly/CHANGELOG.md:7]: no need for the punctuation at the end

File: extensions/recurly/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<p align="center">
+  <img src="assets/recurly-icon.png" height="128">
+  <h1 align="center">Recurly</h1>
+</p>
+
+The extension provides a Recurly integration.
+With it, you can quickly search for user accounts and check their subscriptions.
+
+To obtain an API key, sign in into your Recurly account and go to **Integrations > API Credentials**.
```
Comment (danielsequeira) [inline extensions/recurly/README.md:9]: it would be nice to add an link to the recurly website

**Review by danielsequeira — CHANGES_REQUESTED** [review]
Hi @kudrykv 

Thank you so much for your contribution.

I just went through the PR and added some comments, nothing major.

However I did notice that you are missing the preferences definition in package.json, in order to allow the users to set their Recurly API key and actually use the extension.

**Comment by kudrykv** [comment]
@danielsequeira, thank you for your comment!

> However I did notice that you are missing the preferences definition in package.json, in order to allow the users to set their Recurly API key and actually use the extension

So, this is where Raycast API is limited. On the one hand, Raycast discourages configuring extensions via its commands. On the other hand, there is no (nice) UI in Raycast to do the extensive configuration I need.

I need to add multiple API keys, one per tenant. I need a domain prefix (to build URLs) and a user-friendly name for each API key. At this moment, Raycast doesn't give me the possibility to have some kind of dynamic configuration. I could use a text field and ask users to put the data I need in some predefined format (comma-separated, JSON-formatted, etc), but that is plain unfriendly.

Here, I've broken Raycast's suggestion and went with a separate command (`Tenant configuration`) that allows setting up multiple API keys (last two screenshots). The reason is the UI-friendly setup over requiring serializing data manually.

I'm open to suggestions on how such configuration can be improved. I'm sure I am not the only one who needs the ability to set up a dynamic number of API keys or similar data.

**Review by danielsequeira — CHANGES_REQUESTED** [review]
Hi again @kudrykv,

Sorry again for overseeing what you are trying to achieve with the extension. I just went through the code again and also tested the extension myself and left a few small comments. Mostly nitpicking stuff to keep it consistent with other extensions in our store.

Besides that and for the same reason mentioned above I think the UX and UI of the extension could be improved a bit. I added my comments to the screenshots below:

recurly 2022-07-18 at 16 33 59
recurly 2022-07-18 at 17 02 27
recurly 2022-07-18 at 16 53 38

**Comment by kudrykv** [comment]
@danielsequeira, please take a look again. I've updated commands to perform a right-away search

File: extensions/recurly/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Recurly Changelog
+
+## [Initial Version] - 2022-05-05
```
Comment (danielsequeira) [inline extensions/recurly/CHANGELOG.md:3]:
```suggestion
## [Initial Version] - 2022-07-21
```

**Comment by danielsequeira** [comment]
@kudrykv thank you so much for the patience and all the changes.

Just a few tiny things missing before I merge it.

First could you apply the change I suggested to the CHANGELOG?

Also could you create some new screenshots and replace the existing ones? So they actually showcase the updates you did to the extension.

One last nitpicking suggestion, fine to leave it as it is if you don't want to change this now. But I think in the Search commands it would be a better UI if you didn't have the detail view right column presented when there are no results yet. And just show the no results text and icon in the middle of the screen.
recurly 2022-07-21 at 12 27 40

**Comment by kudrykv** [comment]
> Also could you create some new screenshots and replace the existing ones? So they actually showcase the updates you did to the extension

Yep! Did that; removed one similar screenshot. Are two screenshots enough? I remember seeing an upper limit of six, but I can't find it and whether the lower limit exists.

> didn't have the detail view right column presented when there are no results yet. And just show the no results text and icon in the middle of the screen

Mind = Blown. Didn't think of that. That's a great suggestion, thank you!

**Review by danielsequeira — APPROVED** [review]
All good 🙂 and 2 screenshots are enough. Thank you so much for all the time you invested in this PR!

**Pull request merged**